src/lesson28/js/registration.js

```diff
@@ -45,7 +45,20 @@ const submitButton = document.getElementById('js-submitButton');
 
 submitButton.addEventListener('click', e => {
   e.preventDefault();
-  window.location.href = 'has-registered.html';
+  let users = JSON.parse(localStorage.getItem('users')) || {};
```

Comment:
nits
getItemは`users`がない場合は`null`になるので[Null 合体演算子 (??)](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)を使用する方がより明確だと思いました。
```suggestion
  let users = JSON.parse(localStorage.getItem('users')) ?? {};
```

Reply:
私はちょっと意見が違くて、
左辺がundefinedでもnullでも右辺を返すのは同じなので、
左辺が0や""などfalsyな時にデフォルト値を与えたい場合??を使い、
それらを意識しないでいい時は||を使うことにしています
今回は左辺がそれらを返すことがなく単にnullやundefinedなので||でいいと思っています

Reply:
You can use '??' when you want to handle falsy values, excluding 'null' and 'undefined'.
I learned a lot :）